### ISSUE TITLE
training from ImageNet pretrained model. Miss keys.

### ISSUE THREAD

**Issue by CuiZhiying** [joe-siyuan-qiao/DetectoRS#16]
Hello, thank you very much for your beautiful work.
But when I try to train from ImageNet pretrained model by running
``` Python
python /root/DetectoRS/tools/train.py /root/DetectoRS/configs/DetectoRS/DetectoRS_mstrain_400_1200_r50_40e.py
```
I got a warning which tell me the resnet50 key word miss mathed.
``` Python
unexpected key in source state_dict: fc.weight, fc.bias

missing keys in source state_dict: conv1.weight_gamma, conv1.weight_beta, layer1.0.conv1.weight_gamma, layer1.0.conv1.weight_beta, layer1.0.conv2.weight_gamma, layer1.0.conv2.weight_beta, layer1.0.conv3.weight_gamma, layer1.0.conv3.weight_beta, layer1.0.downsample.0.weight_gamma, layer1.0.downsample.0.weight_beta, layer1.1.conv1.weight_gamma, layer1.1.conv1.weight_beta, layer1.1.conv2.weight_gamma, layer1.1.conv2.weight_beta, layer1.1.conv3.weight_gamma, layer1.1.conv3.weight_beta, layer1.2.conv1.weight_gamma, layer1.2.conv1.weight_beta, layer1.2.conv2.weight_gamma, layer1.2.conv2.weight_beta, layer1.2.conv3.weight_gamma, layer1.2.conv3.weight_beta, layer2.0.conv1.weight_gamma, layer2.0.conv1.weight_beta, layer2.0.conv2.weight_diff, layer2.0.conv2.weight_gamma, layer2.0.conv2.weight_beta, layer2.0.conv2.switch.weight, layer2.0.conv2.switch.bias, layer2.0.conv2.pre_context.weight, layer2.0.conv2.pre_context.bias, layer2.0.conv2.post_context.weight, layer2.0.conv2.post_context.bias, layer2.0.conv2.offset_s.weight, layer2.0.conv2.offset_s.bias, layer2.0.conv2.offset_l.weight, layer2.0.conv2.offset_l.bias, layer2.0.conv3.weight_gamma, layer2.0.conv3.weight_beta, layer2.0.downsample.0.weight_gamma, layer2.0.downsample.0.weight_beta, layer2.1.conv1.weight_gamma, layer2.1.conv1.weight_beta, layer2.1.conv2.weight_diff, layer2.1.conv2.weight_gamma, layer2.1.conv2.weight_beta, layer2.1.conv2.switch.weight, layer2.1.conv2.switch.bias, layer2.1.conv2.pre_context.weight, layer2.1.conv2.pre_context.bias, layer2.1.conv2.post_context.weight, layer2.1.conv2.post_context.bias, layer2.1.conv2.offset_s.weight, layer2.1.conv2.offset_s.bias, layer2.1.conv2.offset_l.weight, layer2.1.conv2.offset_l.bias, layer2.1.conv3.weight_gamma, layer2.1.conv3.weight_beta, layer2.2.conv1.weight_gamma, layer2.2.conv1.weight_beta, layer2.2.conv2.weight_diff, layer2.2.conv2.weight_gamma, layer2.2.conv2.weight_beta, layer2.2.conv2.switch.weight, layer2.2.conv2.switch.bias, layer2.2.conv2.pre_context.weight, layer2.2.conv2.pre_context.bias, layer2.2.conv2.post_context.weight, layer2.2.conv2.post_context.bias, layer2.2.conv2.offset_s.weight, layer2.2.conv2.offset_s.bias, layer2.2.conv2.offset_l.weight, layer2.2.conv2.offset_l.bias, layer2.2.conv3.weight_gamma, layer2.2.conv3.weight_beta, layer2.3.conv1.weight_gamma, layer2.3.conv1.weight_beta, layer2.3.conv2.weight_diff, layer2.3.conv2.weight_gamma, layer2.3.conv2.weight_beta, layer2.3.conv2.switch.weight, layer2.3.conv2.switch.bias, layer2.3.conv2.pre_context.weight, layer2.3.conv2.pre_context.bias, layer2.3.conv2.post_context.weight, layer2.3.conv2.post_context.bias, layer2.3.conv2.offset_s.weight, layer2.3.conv2.offset_s.bias, layer2.3.conv2.offset_l.weight, layer2.3.conv2.offset_l.bias, layer2.3.conv3.weight_gamma, layer2.3.conv3.weight_beta, layer3.0.conv1.weight_gamma, layer3.0.conv1.weight_beta, layer3.0.conv2.weight_diff, layer3.0.conv2.weight_gamma, layer3.0.conv2.weight_beta, layer3.0.conv2.switch.weight, layer3.0.conv2.switch.bias, layer3.0.conv2.pre_context.weight, layer3.0.conv2.pre_context.bias, layer3.0.conv2.post_context.weight, layer3.0.conv2.post_context.bias, layer3.0.conv2.offset_s.weight, layer3.0.conv2.offset_s.bias, layer3.0.conv2.offset_l.weight, layer3.0.conv2.offset_l.bias, layer3.0.conv3.weight_gamma, layer3.0.conv3.weight_beta, layer3.0.downsample.0.weight_gamma, layer3.0.downsample.0.weight_beta, layer3.1.conv1.weight_gamma, layer3.1.conv1.weight_beta, layer3.1.conv2.weight_diff, layer3.1.conv2.weight_gamma, layer3.1.conv2.weight_beta, layer3.1.conv2.switch.weight, layer3.1.conv2.switch.bias, layer3.1.conv2.pre_context.weight, layer3.1.conv2.pre_context.bias, layer3.1.conv2.post_context.weight, layer3.1.conv2.post_context.bias, layer3.1.conv2.offset_s.weight, layer3.1.conv2.offset_s.bias, layer3.1.conv2.offset_l.weight, layer3.1.conv2.offset_l.bias, layer3.1.conv3.weight_gamma, layer3.1.conv3.weight_beta, layer3.2.conv1.weight_gamma, layer3.2.conv1.weight_beta, layer3.2.conv2.weight_diff, layer3.2.conv2.weight_gamma, layer3.2.conv2.weight_beta, layer3.2.conv2.switch.weight, layer3.2.conv2.switch.bias, layer3.2.conv2.pre_context.weight, layer3.2.conv2.pre_context.bias, layer3.2.conv2.post_context.weight, layer3.2.conv2.post_context.bias, layer3.2.conv2.offset_s.weight, layer3.2.conv2.offset_s.bias, layer3.2.conv2.offset_l.weight, layer3.2.conv2.offset_l.bias, layer3.2.conv3.weight_gamma, layer3.2.conv3.weight_beta, layer3.3.conv1.weight_gamma, layer3.3.conv1.weight_beta, layer3.3.conv2.weight_diff, layer3.3.conv2.weight_gamma, layer3.3.conv2.weight_beta, layer3.3.conv2.switch.weight, layer3.3.conv2.switch.bias, layer3.3.conv2.pre_context.weight, layer3.3.conv2.pre_context.bias, layer3.3.conv2.post_context.weight, layer3.3.conv2.post_context.bias, layer3.3.conv2.offset_s.weight, layer3.3.conv2.offset_s.bias, layer3.3.conv2.offset_l.weight, layer3.3.conv2.offset_l.bias, layer3.3.conv3.weight_gamma, layer3.3.conv3.weight_beta, layer3.4.conv1.weight_gamma, layer3.4.conv1.weight_beta, layer3.4.conv2.weight_diff, layer3.4.conv2.weight_gamma, layer3.4.conv2.weight_beta, layer3.4.conv2.switch.weight, layer3.4.conv2.switch.bias, layer3.4.conv2.pre_context.weight, layer3.4.conv2.pre_context.bias, layer3.4.conv2.post_context.weight,
```

**Comment by joe-siyuan-qiao** [comment]
Thanks for the question. That is normal. They are not in the pre-trained models. They will be trained by DetectoRS.

**Comment by CuiZhiying** [comment]
OK~ thanks for you quick reply.
Another question, I found that the learning rate is 0.02 while the batchsize is 1 for each GPU. I wonder how many GPUs you use to train the whole model?

**Comment by Mozen** [comment]
@joe-siyuan-qiao my GPU is TITAN XP 12G，and image size is fixed with (1333,800), it will mermory error with config  ResNext32x4.  how to reduce the mermory?

**Comment by joe-siyuan-qiao** [comment]
@CuiZhiying We used 4 GPUs and 8 GPUs with 2 imgs/GPU and 1 img/GPU. Please adjust the learning rate according to the effective batch size. The provided config files would require 16 GPUs if the learning rate is 0.02 and the batch size is 1.

**Comment by joe-siyuan-qiao** [comment]
@Mozen if memory is the bottleneck, please consider setting _with_cp_ in the backbone to true to reduce memory usage. You can also consider waiting for us transferring DetectoRS to mmdetection v2.0 where more detectors with less memory requirement will be supported.

**Comment by Mozen** [comment]
@joe-siyuan-qiao ok, thinks for you reply.
